### PR TITLE
KFSPTS-21562: Remove CU customization for the CuNextSecurityModelDefinitionIdFinder class

### PR DESCRIPTION
**Pull request by nkimble**

### CODE REVIEW

File: src/main/resources/edu/cornell/kfs/sec/cu-spring-sec.xml [hunk 1]
```diff
@@ -1,33 +0,0 @@
...
-    xmlns:p="http://www.springframework.org/schema/p"
-    xsi:schemaLocation="http://www.springframework.org/schema/beans
-        http://www.springframework.org/schema/beans/spring-beans-2.0.xsd">    
-
-    <!--
-        The Access Security module configuration bean hierarchy has two levels of parent beans.
-        We override the one higher up in the path, to minimize required changes and to prevent
-        reloading of the whole SEC module (which could mess up our other modules' bean overrides).
-     -->
-    <bean id="secModule-parentBean" parent="secModule-coreOnly-parentBean" abstract="true">
```
Comment (cah292) [inline src/main/resources/edu/cornell/kfs/sec/cu-spring-sec.xml:25]: If CuSecurityModelMaintainableImpl is being kept, then this bean needs to be kept as well.

File: src/main/resources/edu/cornell/kfs/sec/document/datadictionary/SecurityModelMaintenanceDocument.xml [hunk 2]
```diff
@@ -1,49 +0,0 @@
...
-   - but WITHOUT ANY WARRANTY; without even the implied warranty of
-   - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
-   - GNU Affero General Public License for more details.
-   - 
-   - You should have received a copy of the GNU Affero General Public License
-   - along with this program.  If not, see <http://www.gnu.org/licenses/>.
- -->
-<beans xmlns="http://www.springframework.org/schema/beans" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:p="http://www.springframework.org/schema/p" xsi:schemaLocation="http://www.springframework.org/schema/beans         http://www.springframework.org/schema/beans/spring-beans-2.0.xsd">
-
-    <bean id="SecurityModelMaintenanceDocument" parent="SecurityModelMaintenanceDocument-parentBean"
```
Comment (cah292) [inline src/main/resources/edu/cornell/kfs/sec/document/datadictionary/SecurityModelMaintenanceDocument.xml:22]: If CuSecurityModelMaintainableImpl is being kept, then this bean needs to be kept as well. (All the other beans in this file can be removed, though.)

File: src/main/resources/institutional-config.properties [hunk 3]
```diff
@@ -47,7 +47,6 @@ cu.spring.source.files=\
   classpath:edu/cornell/kfs/module/cg/cu-spring-cg.xml,\
   classpath:edu/cornell/kfs/gl/cu-spring-gl.xml,\
   classpath:edu/cornell/kfs/fp/cu-spring-fp.xml,\
-  classpath:edu/cornell/kfs/sec/cu-spring-sec.xml,\
```
Comment (cah292) [inline src/main/resources/institutional-config.properties:50]: If CuSecurityModelMaintainableImpl is being kept, then this "cu-spring-sec" entry should be kept in place.

**Review by cah292 — CHANGES_REQUESTED** [review]
I've finished my initial review, and I agree that the model-def-ID-finder customization can be removed, since KualiCo's related OJB setup will generate a new ID for this object from the appropriate DB sequence.

However, I noticed that this PR is affecting an unrelated role-ID-related customization that should be kept in place. See my comments below for details.

File: src/main/resources/institutional-config.properties (hunk 3 above)
Comment (nkimble) [reply src/main/resources/institutional-config.properties:50]: fixed

File: src/main/java/edu/cornell/kfs/sec/document/CuSecurityModelMaintainableImpl.java [hunk 4]
```diff
@@ -1,13 +0,0 @@
-package edu.cornell.kfs.sec.document;
-
-import org.kuali.kfs.sec.businessobject.SecurityModel;
-import org.kuali.kfs.sec.document.SecurityModelMaintainableImpl;
-
-public class CuSecurityModelMaintainableImpl extends SecurityModelMaintainableImpl {
```
[the remark this answers is not in the document]
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/sec/document/CuSecurityModelMaintainableImpl.java]: fixed

File: src/main/resources/edu/cornell/kfs/sec/cu-spring-sec.xml (hunk 1 above)
Comment (nkimble) [reply src/main/resources/edu/cornell/kfs/sec/cu-spring-sec.xml]: restored file

File: src/main/resources/edu/cornell/kfs/sec/document/datadictionary/SecurityModelMaintenanceDocument.xml (hunk 2 above)
Comment (nkimble) [reply src/main/resources/edu/cornell/kfs/sec/document/datadictionary/SecurityModelMaintenanceDocument.xml:22]: fixed

**Review by nkimble — COMMENTED** [review]
Thanks for catching this Chad. Changes have been made and should be ready for review after the unit tests complete.

**Review by cah292 — APPROVED** [review]
Changes look good, unit tests pass, and my concerns have been addressed. Merging.

**Pull request merged**